.pre-commit-config.yaml

```diff
@@ -0,0 +1,64 @@
+# https://pre-commit.com/
+repos:
+  # General hooks
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
+    hooks:
+      - id: check-merge-conflict
+      - id: end-of-file-fixer
+      - id: mixed-line-ending
+      - id: trailing-whitespace
+        args: ["--markdown-linebreak-ext=md"]
+  - repo: https://github.com/pre-commit/mirrors-prettier
+    rev: v2.7.1
+    hooks:
+      - id: prettier
+        name: doc format
+        files: ^doc/
+  # Frontend hooks
+  - repo: https://github.com/pre-commit/mirrors-prettier
+    rev: v2.7.1
+    hooks:
+      - id: prettier
+        name: frontend format
+        args: [--config=frontend/prettier.config.js, --ignore-path=frontend/.prettierignore]
+        files: ^frontend/
+        types_or:
+          - javascript
+          - jsx
+          - ts
+          - tsx
+          - json
+          - css
+          - scss
+          - markdown
+        additional_dependencies:
+          - prettier@2.7.1
+          - prettier-plugin-tailwindcss@0.2.1
+  - repo: https://github.com/pre-commit/mirrors-eslint
+    rev: v8.24.0
+    hooks:
+      - id: eslint
+        name: frontend lint
+        args: [--config=frontend/.eslintrc.json, --max-warnings=0]
+        verbose: true
+        files: ^frontend/
+        types: [file]
+        types_or:
+          - javascript
+          - jsx
+          - ts
+          - tsx
+        additional_dependencies:
+          - eslint@8.24.0
+          - eslint-plugin-react@7.31.8
+          - eslint-plugin-react-hooks@4.6.0
+  # Backend hooks
+  - repo: https://github.com/doublify/pre-commit-rust
+    rev: v1.0
+    hooks:
+      - id: fmt
+        name: backend format
+        args: [--manifest-path=backend/Cargo.toml]
+        files: ^backend/
+        pass_filenames: false
```

Comment:
Please give a bit of information about which config you changed (and why?).

Reply:
I'm not sure what you mean.
I didn't change any config for the backend formatting.
All I did was tell pre-commit how to call `cargo fmt` on the backend subfolder only.

Reply:
I mean that you explain e.g. what `pass_filenames` is and does (or why it is false here) as inline-comment.

Reply:
Sure.
1. I need to pass `args: [--manifest-path=backend/Cargo.toml]` because cargo is called from the root project dir which doesn't include `Cargo.toml` for the backend.
2. The `files: ^backend/` option tells pre-commit to skip this hook if no files in the backend dir is staged.
3. `pass_filenames: falsepass_filenames: false` tells pre-commit not to pass the staged & matched file names to the executed command, `cargo fmt`. I did this because the command doesn't take any file name arguments. It would complain and not run properly without it.